### ISSUE TITLE
Multiple upgraded elements causes breaks

### ISSUE THREAD

**Issue by philcchen** [ampproject/worker-dom#158]
Not sure if this currently allowed or not, but it seems calling `upgradeElement` on multiple elements can cause issues, presumably due to the state that is being shared between the upgraded components.

One example of this is the strings map here: https://github.com/ampproject/worker-dom/blob/master/src/main-thread/strings.ts#L18 which causes there to be an index-mismatch here https://github.com/ampproject/worker-dom/blob/master/src/main-thread/nodes.ts#L68

Repro (in the React map demo here:  https://github.com/ampproject/worker-dom/blob/master/demo/react-map/index.html )
Modify the body to be 
```
<body>
  <div src="dist/main.js" id="upgrade-me"></div>
  <div src="dist/main.js" id="upgrade-me-2"></div>
  <script type="module">
    import {upgradeElement} from '/dist/index.mjs';
    upgradeElement(document.getElementById('upgrade-me'), '/dist/worker.mjs');
    upgradeElement(document.getElementById('upgrade-me-2'), '/dist/worker.mjs');
  </script>
  <script nomodule async=false defer>
    document.addEventListener('DOMContentLoaded', function() {
      MainThread.upgradeElement(document.getElementById('upgrade-me'), '/dist/worker.js');
      MainThread.upgradeElement(document.getElementById('upgrade-me-2'), '/dist/worker.js');
    }, false);
  </script>
</body>
```

**Comment by kristoferbaxter** [comment]
It's intended to allow multiple root nodes to be delegated to the worker. Looks like a series of bugs to fix!

**Comment by kristoferbaxter** [comment]
This was resolved in the `0.3.0` release. Please let us know if you continue to see the issue (also check the example of multiple roots in the demo folder).
